### ISSUE TITLE
Use mimeapps.list instead of defaults.list in file association config

### ISSUE THREAD

**Issue by jleclanche** [lxde/lxqt#11]
Razor-qt/razor-qt#603

**Comment by pmattern** [comment]
Didn't look through the commits thoroughly, but according to my tests lxqt-config-file-associations from latest lxqt-config is writing into $XDG_DATA_HOME/applications/mimeapps.list and hence not into defaults.list any more.
(On a side note I failed to figure out from where it's taking defaults as long as this file doesn't exist.)

Aside from this, none of the other LXQt components involved in default applications seems to make use of defaults.list either: lxqt-config-session is using environment variables anyway. libfm-pref-apps is writing into $XDG_CONFIG_HOME/mimeapps.list, customized applications go to $XDG_DATA_HOME/applications/userapp-foo-bar.desktop.

Issue can thus be closed, imo.

**Comment by surlykke** [comment]
2015-02-20 18:12 GMT+01:00 Peter Mattern <notifications@github.com>:

> Didn't look through the commits thoroughly, but according to my tests
> lxqt-config-file-associations from latest lxqt-config is writing into
> $XDG_DATA_HOME/applications/mimeapps.list and hence not into defaults.list
> any more.
>
Yeah, I think I implemented this some months ago. Sorry if I neglected to
close the issue.


> (On a side note I failed to figure out from where it's taking defaults as
> long as this file doesn't exist.)
>
If no entry is found in mimeapps.list, we find those desktopfiles that
declare the abillity to handle the mimetype in case. These desktopfiles can
have an attribute 'InitialPreference'. Desktopfiles with higher
InitialPreference are preferred. If several desktopfiles have the same
InitialPreference it is random which is taken. This is what can be deduced
from the standard AFAIR.

br. Chr.
